### PR TITLE
[Yaml] Allow comments after tab

### PR DESCRIPTION
**Pull request by superdav42**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |

If a yml file has a tab character before a line ending comment the comment will be included in the parsed value. Yaml spec allows tab or space as whitespace characters so we need to check for tab as well. See included test.
Recently caused an odd and hard to find bug in our project.

### CODE REVIEW

File: src/Symfony/Component/Yaml/Inline.php [hunk 1]
```diff
@@ -209,7 +209,8 @@ public static function parseScalar($scalar, $delimiters = null, $stringDelimiter
                 $i += strlen($output);
 
                 // remove comments
-                if (false !== $strpos = strpos($output, ' #')) {
+                if ((false !== $strpos = strpos($output, '#')) && 
+                    ($output[$strpos-1] == ' ' || $output[$strpos-1] == "\t")) {
```
Comment (fabpot) [inline src/Symfony/Component/Yaml/Inline.php]: should be on one line

**Comment by superdav42** [comment]
@fabpot Fixed the code style.

File: src/Symfony/Component/Yaml/Inline.php [hunk 2]
```diff
@@ -209,7 +209,7 @@ public static function parseScalar($scalar, $delimiters = null, $stringDelimiter
                 $i += strlen($output);
 
                 // remove comments
-                if (false !== $strpos = strpos($output, ' #')) {
+                if ((false !== $strpos = strpos($output, '#')) && ($output[$strpos - 1] == ' ' || $output[$strpos - 1] == "\t")) {
```
Comment (WouterJ) [inline src/Symfony/Component/Yaml/Inline.php]: you have to use `===` and the yoda style (`' ' === $output[$strpos - 1]`)

**Comment by fabpot** [comment]
@superdav42 Can you also rebase (instead of merge) to get rid of the merge commits?

**Comment by fabpot** [comment]
Can you also paste the part of the YAML configuration that talks about this? Thanks.

**Comment by superdav42** [comment]
@fabpot Rebasing doesn't seem to work so well after publishing, Created new PR https://github.com/symfony/symfony/pull/15793

**Comment by stof** [comment]
@superdav42 you just need to force the push to your feature branch (as explained in our documentation btw). For branches used for PRs, it is fine

**Pull request closed without merging**